std/fs/write_json_test.ts

```diff
@@ -1,25 +1,18 @@
 // Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
-import {
-  assertEquals,
-  assertThrowsAsync,
-  assertThrows,
-} from "../testing/asserts.ts";
 import * as path from "../path/mod.ts";
+import { assertEquals } from "../testing/asserts.ts";
 import { writeJson, writeJsonSync } from "./write_json.ts";
 
 const testdataDir = path.resolve("fs", "testdata");
 
 Deno.test("writeJsonIfNotExists", async function (): Promise<void> {
   const notExistsJsonFile = path.join(testdataDir, "file_not_exists.json");
 
-  await assertThrowsAsync(
-    async (): Promise<void> => {
-      await writeJson(notExistsJsonFile, { a: "1" });
-      throw new Error("should write success");
-    },
-    Error,
-    "should write success",
-  );
+  try {
+    await writeJson(notExistsJsonFile, { a: "1" });
+  } catch {
+    // empty
+  }
```

Comment:
Yes the previous code was really strange. Why did you replace with a `try/catch` block?
```suggestion
  await writeJson(notExistsJsonFile, { a: "1" });
```

Reply:
> Why did you replace with a `try/catch` block?

To preserve the previous behavior. Based on review of the previous code, this is my understanding of what the tests were doing:

- Write an empty file to the test data directory
- Invoke `writeJson` to overwrite that file with the test data, but don't stop the test if it throws (so the file can still be deleted during cleanup)
- Read the content from the file
- Delete the file
- Compare the content of the file to the expected data using `assertEquals` (this is where the real "test" is happening)

If cleanup of the test files is not important, then the `try/catch` blocks can be removed (but I think this is useful for maintaining repo state).

---

For the specific test you highlighted, I think it makes sense not to use the `try/catch` becuase it is the first write of the file, and if it throws then cleanup will not be necessary. However, in reference to the change in @ry's [commit](https://github.com/denoland/deno/pull/6889/commits/ee865cff4d97599d997edef9fd61be0ec48fe06a): the sync test ([`writeJsonSyncIfNotExists`](https://github.com/denoland/deno/pull/6889/files#diff-7b1522a54acee65b2582a1dda5c897e2R121)) should mirror that change for consistency.

Reply:
> but don't stop the test if it throws

That wasn't the previous behaviour. If `writeJson()` failed, the containing `assertThrows()` would fail because it wouldn't have the correct error message. I don't think any of these is doing any cleanup or you'd see a `try/finally`.

Reply:
> I don't think any of these is doing any cleanup or you'd see a `try/finally`.

That's how I'd do test cleanup also since Deno doesn't provide setup/teardown methods.

If you don't think any cleanup is happening, what do you think is the purpose of the `Deno.remove` statements that were in the tests?

Reply:
> If you don't think any cleanup is happening, what do you think is the purpose of the `Deno.remove` statements that were in the tests?

Oops, I meant fault-tolerant cleanup.

Reply:
@nayeemrmn How do you read the intent of the previous tests?